src/utils.js

```diff
@@ -224,6 +226,44 @@ export function mergeObjects(obj1, obj2, concatArrays = false) {
   }, acc);
 }
 
+function objectNeedsTrimmed(obj) {
+  // Recursively determine if an object needs to be simplified to `undefined`.
+  return Object.keys(obj).every(key => {
```

Comment:
Shouldn't we use `some` instead of `every` here?

Reply:
I probably needed to do a better job explaining this in the PR, but we need to convert objects with all of their fields set to `undefined` to simply `undefined` and do this recursively. For example, convert

`{ "a": true, "b": { "c": undefined, "d": { "e": undefined} } }` to `{"a": true, "b": undefined}`

The reason is, assuming we have a schema where `b` is optional but its fields are required, if `b` is set to an object the validator requires all of its required fields to be set. So we must "trim" down its unspecified fields and `b` itself to simply `undefined`. The `every` is requiring all of an objects fields are `undefined` to do a "trim". I do not believe setting it to `some` would work because the user would be unable to set the fields of an object (the entire object would keep being set to `undefined`) unless they were somehow able to simultaneously set all fields at once. Thanks for the feedback!